tests/unit/commands/test_lifecycle.py

```diff
@@ -79,6 +108,31 @@ def run_managed(self, parsed_args: argparse.Namespace) -> bool:  # noqa: ARG002
     return FakeCommand
 
 
+@pytest.mark.parametrize(
+    ("ua_is_attached", "ua_enabled_services", "proservices_args", "expected_exception"),
+    PRO_SERVICE_CONFIGS,
+)
+def test_validate_pro_services(
+    uaclient_mock,
+    ua_is_attached,
+    ua_enabled_services,
+    proservices_args,
+    expected_exception,
+):
+    # unpack fixture
+    mock_uaclient_module, set_mock_uaclient = uaclient_mock
+
+    try:
+        # configure api state
+        set_mock_uaclient(ua_is_attached, ua_enabled_services)
+
+        # create and validate pro services
+        proservices = ProServices(proservices_args)
+        proservices.validate()
```

Comment:
This will likely assert None. Lets double check here

Reply:
Resolved by using a variable context manager. We use pytest.raises if we pass an exception to catch, otherwise we user the nullcontext context manager.